pkg/engine/runtime/kubernetes_runtime.go

```diff
@@ -116,20 +117,21 @@ func (k *KubernetesRuntime) Apply(ctx context.Context, request *ApplyRequest) *A
 			}
 			if patchedObj, err := resource.Patch(ctx, planObj.GetName(), types.MergePatchType, patchBody, patchOptions); err == nil {
 				res = patchedObj
-				tryClientSide = false
 			} else {
-				log.Errorf("ServerSideDryRun Patch failed, err: %v", err)
-			}
-		}
-		if tryClientSide {
-			// Fall back to ClientSideDryRun
-			mergedPatch, err := jsonpatch.MergePatch([]byte(current), patchBody)
-			if err != nil {
-				return &ApplyResponse{nil, status.NewErrorStatus(err)}
-			}
-			res = &unstructured.Unstructured{}
-			if err = res.UnmarshalJSON(mergedPatch); err != nil {
-				return &ApplyResponse{nil, status.NewErrorStatus(err)}
+				// Fall back to ClientSideDryRun
+				log.Errorf("ServerSideDryRun patch %s failed, fall back to ClientSideDryRun; err: %v", planState.ID, err)
+
+				// Merge 3-way patch
+				mergedPatch, err := jsonpatch.MergePatch([]byte(current), patchBody)
```

Comment:
Why do we need another merge here since `patchBody` already ThreeWayJSONMergePatch before in L91

Reply:
> Why do we need another merge here since `patchBody` already ThreeWayJSONMergePatch before in L91

`ThreeWayJSONMergePatch()` will return a patch body, not a patch result.